src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores-export-button.component.ts

```diff
@@ -0,0 +1,76 @@
+import { StudentParticipation } from 'app/entities/participation/student-participation.model';
+import { round } from 'app/shared/util/utils';
+import { JhiAlertService } from 'ng-jhipster';
+import { ProgrammingExerciseStudentParticipation } from 'app/entities/participation/programming-exercise-student-participation.model';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { Injectable, Component, Input } from '@angular/core';
+
+import { ResultService } from 'app/exercises/shared/result/result.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+
+@Component({
+    selector: 'jhi-exercise-scores-export-button',
+    template: `
+        <button class="btn btn-info btn-sm mr-1" (click)="exportResults()">
+            <fa-icon [icon]="'download'"></fa-icon>
+            <span class="d-none d-md-inline" jhiTranslate="artemisApp.exercise.exportResults">Export Results</span>
+        </button>
+    `,
+})
+@Injectable({ providedIn: 'root' })
+export class ExerciseScoresExportButtonComponent {
+    @Input() exercises: Exercise[] = []; // Used to export multiple scores together
+    @Input() exercise: Exercise | ProgrammingExercise;
+
+    constructor(private resultService: ResultService, private jhiAlertService: JhiAlertService) {}
+
+    /**
+     * Exports the exercise results as a csv file
+     */
+    exportResults() {
+        if (this.exercises.length === 0 && this.exercise !== undefined) {
+            this.exercises = this.exercises.concat(this.exercise);
+        }
+        this.exercises.forEach((exercise) => {
+            this.exercise = exercise;
+            this.resultService.getResults(exercise).subscribe((data) => {
+                const rows: string[] = [];
+                const results = data.body || [];
+                if (results.length === 0) {
+                    this.jhiAlertService.warning(`artemisApp.exercise.exportResultsEmptyError`, { exercise: exercise.title });
+                    window.scroll(0, 0);
+                    return;
+                }
+                results.forEach((result, index) => {
+                    const studentParticipation = result.participation! as StudentParticipation;
+                    const { participantName, participantIdentifier } = studentParticipation;
+                    const score = round(result.score);
+
+                    if (index === 0) {
+                        const nameAndUserNameColumnHeaders = studentParticipation.team ? 'Team Name,Team Short Name' : 'Name,Username';
+                        const optionalStudentsColumnHeader = studentParticipation.team ? ',Students' : '';
```

Comment:
This check is repeated 3 times in the code and the same thing for exercise.type !== PROGRAMMING, I think it would be better if you can group them.

Reply:
I think readability would suffer from a grouping of these two checks.. Or at least as I would refactor it..